addons/mrp/models/stock_picking.py

```diff
@@ -106,6 +110,31 @@ def action_detailed_operations(self):
         action['context']['has_kits'] = self.has_kits
         return action
 
+    @api.depends('group_id.stock_move_ids.created_production_id.procurement_group_id.mrp_production_ids',
+                'move_ids.move_dest_ids.raw_material_production_id')
+    def _compute_mrp_production_ids(self):
+        for record in self:
+            mo_ids = self.env['mrp.production'].search([('procurement_group_id', '=', record.group_id.id)])
+            mo_ids |= record.move_ids.move_dest_ids.raw_material_production_id
+            record.mrp_production_count = len(mo_ids)
+
+    def action_view_mrp_production(self):
+        self.ensure_one()
+        mrp_production_ids = self.env['mrp.production'].search([('procurement_group_id', '=', self.group_id.id)])
+        mrp_production_ids |= self.move_ids.move_dest_ids.raw_material_production_id
```

Comment:
I would use 2 computed fields (production_ids and production_count (rename). not stored and in the same compute